File: app/src/main/java/workshop1024/com/xproject/controller/fragment/home/news/SearchNewsFragment.java
package workshop1024.com.xproject.controller.fragment.home.news;

import android.os.Bundle;

import workshop1024.com.xproject.R;
import workshop1024.com.xproject.controller.fragment.SubFragment;

public class SearchNewsFragment extends NewsListFragment implements SubFragment {
    private static final String SEARCH_NAME = "search_Name";
    private String mFilterName;

    public static SearchNewsFragment newInstance(String searchName) {
        SearchNewsFragment searchNewsFragment = new SearchNewsFragment();
        searchNewsFragment.setMNavigationItemId(R.id.leftnavigator_menu_home);
        Bundle args = new Bundle();
        args.putString(SEARCH_NAME, searchName);
        searchNewsFragment.setArguments(args);
        return searchNewsFragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            mFilterName = getArguments().getString(SEARCH_NAME);
        }
    }

    @Override
    public void getNewsList() {
        mNewsRepository.getNewsListBySearch(mFilterName, this);
    }
}

File: app/src/main/java/workshop1024/com/xproject/controller/fragment/home/news/SearchNewsFragment.kt
package workshop1024.com.xproject.controller.fragment.home.news

import android.os.Bundle

import workshop1024.com.xproject.R
import workshop1024.com.xproject.controller.fragment.SubFragment

class SearchNewsFragment : NewsListFragment(), SubFragment {
    private var mFilterName: String? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (arguments != null) {
            mFilterName = arguments!!.getString(SEARCH_NAME)
        }
    }

    override fun getNewsList() {
        mNewsRepository.getNewsListBySearch(mFilterName!!, this)
    }

    companion object {
        private const val SEARCH_NAME = "search_Name"

        fun newInstance(searchName: String): SearchNewsFragment {
            val searchNewsFragment = SearchNewsFragment()
            searchNewsFragment.mNavigationItemId = R.id.leftnavigator_menu_home
            val args = Bundle()
            args.putString(SEARCH_NAME, searchName)
            searchNewsFragment.arguments = args
            return searchNewsFragment
        }
    }
}
